test/bitarray.jl

```diff
@@ -128,11 +140,48 @@ for n in [1; 1023:1025]
     end
 end
 
-@test_throws BoundsError size(trues(5),0)
+@test_throws BoundsError size(trues(5), 0)
+
+# reshape and resize!
+
+let b1 = bitrand(n1, n2)
+    @check_bit_operation reshape(b1, (n2,n1)) BitMatrix
+    @test_throws DimensionMismatch reshape(b1, (1,n1))
+end
+let b1 = bitrand(s1, s2, s3, s4)
+    @check_bit_operation reshape(b1, (s3,s1,s2,s4)) BitArray{4}
+    @test_throws DimensionMismatch reshape(b1, (1,n1))
+end
+
+let b1 = bitrand(v1)
+    @test_throws BoundsError resize!(b1, -1)
+    @check_bit_operation resize!(b1, v1 ÷ 2) BitVector
+    gr(b) = (resize!(b, v1)[(v1÷2):end] = 1; b)
+    @check_bit_operation gr(b1) BitVector
+end
+
+# sizeof (issue #7515)
+@test sizeof(BitArray(64)) == 8
+@test sizeof(BitArray(65)) == 16
 
 timesofar("utils")
 
-## Constructors from iterables ##
+## Constructors ##
+
+# non-Int dims constructors
+
+let b1 = BitArray(Int32(v1)),
+    b2 = BitArray(Int64(v1))
+    @test size(b1) == size(b2)
```

Comment:
previous version of this test was also checking equality of the bitarrays to one another, not just their sizes